src/librustc_metadata/csearch.rs

```diff
@@ -140,6 +140,31 @@ impl<'tcx> CrateStore<'tcx> for cstore::CStore {
         })
     }
 
+    /// If possible, this returns a global path to the external item `def` that is visible from
+    /// at least one local module. If the external item's crate was declared with an
+    /// `extern crate`, the path is guarenteed to use the `extern crate`.
+    fn visible_item_path(&self, def: DefId) -> Option<Vec<hir_map::PathElem>> {
```

Comment:
It might be simpler to return item_path() rather than None if there is no visible item path